hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/TransformStatefulP.java

```diff
@@ -154,6 +167,9 @@ void reset(Watermark wm) {
 
     @Override
     public boolean saveToSnapshot() {
+        if (inComplete) {
```

Comment:
this needs more explanation, like in SessionWindowP. Do we perhaps need better support for this in ProcessorTasklet?